### ISSUE TITLE
Ability to ignore a steps ?

### ISSUE THREAD

**Issue by schhor** [authorjapps/zerocode#339]
Hi, 

I'm just discovered zero code and it is great.

I would like to know if there is a way to disable some steps from a scenario ? (for example by adding a attribut ignore: true ).

This ability could help me to do some "BabySteps coding". I would like to code in a T.D.D. style approach with Baby steps.

Here the process I like to follow :
- Scenario is written in json and commit in a feature branch.
- Scenario is review with P.R. and then merge into master (json file is not referenced in Junit Test File, so It won't be launch)
- When implementing the code matching the scenario, all steps of the scenario are made disable except those that will be implements by the code, then iterate until all steps is active.

**Comment by nirmalchandra** [comment]
@schhor , `ignoreStepFailures` is already present, but this may be in a slight different context.
Please have a look [here](https://github.com/authorjapps/zerocode/wiki#ignoring-step-failures). This applies to the **entire scenario**, bot individual steps.

> I would like to know if there is a way to disable some steps from a scenario ? (for example by adding a attribute ignore: true ).

Yes, we can sort this out for you in the upcoming release.  This is bit granular and you want the control in the individual steps. Sounds good 👍

**Comment by schhor** [comment]
Hi @nirmalchandra,

I have seen the ignoreStepFailures but like you have guess I don't want my build to fail.

It would be great if this can be done in the next release !  Thanks you so much :)

**Comment by schhor** [comment]
A another thing, can I contribute ? do you have any entry point to know how to to do? thanks

**Comment by authorjapps** [comment]
Note-
This is not kind of or related to issue #328 
Can you please unlink(when you get a chance)?

**Comment by schhor** [comment]
Hi,
I have edited my comment to remove the link

**Comment by authorjapps** [comment]
Thanks @schhor. Apologies for the delay in getting this fixed.

This will be available in `v 1.3.17`.

_Due to end-of-year and X-mas time, we are experiencing a slight delay in the review and release processes.  We are trying our best to get the fixes released by early 2020._

**Comment by vkjha2000** [comment]
> Hi,
> 
> I'm just discovered zero code and it is great.
> 
> I would like to know if there is a way to disable some steps from a scenario ? (for example by adding a attribute ignore: true ).
> 
> This ability could help me to do some "BabySteps coding". I would like to code in a T.D.D. style approach with Baby steps.
> 
> Here the process I like to follow :
> 
> * Scenario is written in json and commit in a feature branch.
> * Scenario is review with P.R. and then merge into master (json file is not referenced in Junit Test File, so It won't be launch)
> * When implementing the code matching the scenario, all steps of the scenario are made disable except those that will be implements by the code, then iterate until all steps is active.

@schhor ,
I am one of the initial designer of the HTTP DSLs.

Just trying to understand the test-flow, before we actually implement it. 
Can you please validate ?

Initially, you would have all `ignored` steps as below:
```java
[{
		"step": "step1",
		"ignore": true,
		"method": "POST"
                ...
	},
	{
		"step": "step2",
		"ignore": true,
		"method": "GET"
                ...
	}
]
```

then once you implement the `POST` method(mark step1 as false):
```java
[{
		"step": "step1",
		"ignoreStep": false, //or skip this field. Default is false.
		"method": "POST"

	},
	{
		"step": "step2",
		"ignoreStep": true,
		"method": "GET"
	}
]
```
next, after you implemented the `GET` method(mark both steps as false):
```java
[{
		"step": "step1",
		"ignoreStep": false,
		"method": "POST"

	},
	{
		"step": "step2",
		"ignoreStep": false,
		"method": "GET"
	}
]
```

**Comment by authorjapps** [comment]
> A another thing, can I contribute ? do you have any entry point to know how to to do? thanks

@schhor , Thanks for your interest.
assuming @vkjha2000 's proposal is what you are looking for, you can check this method

```java
org.jsmart.zerocode.core.runner.ZeroCodeMultiStepsScenarioRunnerImpl#executeSteps
```

Then, check the `isIgnored()` flag is false, then do not execute this step.
> This sounds simpler, but we are open to any other approach you think is great? Also, please go through the "CONTRIBUTING.md file" for the guidelines.

```java
for (Step thisStep : parameterizedScenario.getSteps()) {
            ...
            if(thisStep.isIgnored()){ 
                 continue;
            }
            ...
}
```

**Comment by schhor** [comment]
Hi @vkjha2000, 

this is exactly that I want to do.

thanks

**Comment by authorjapps** [comment]
To be released in `V1.3.9`
